lib/haj_web/live/group_admin_live.ex

```diff
@@ -2,29 +2,40 @@ defmodule HajWeb.GroupAdminLive do
   use HajWeb, :live_view
 
   alias Haj.Spex
-  # Old stuff, should be redone
+  alias Haj.Repo
 
-  def mount(_params, %{"user_token" => token, "show_group_id" => id}, socket) do
-    show_group = Haj.Spex.get_show_group!(id)
+  def mount(%{"show_group_id" => show_group_id}, _session, socket) do
+    show_group = Spex.get_show_group!(show_group_id)
     changeset = Spex.change_show_group(show_group)
-
-    socket =
-      socket
-      |> assign_new(:current_user, fn ->
-        Haj.Accounts.get_user_by_session_token(token) |> Haj.Spex.preload_user_groups()
-      end)
-      |> assign(:show_group, show_group)
-      |> assign(
-        changeset: changeset,
-        query: nil,
-        loading: false,
-        matches: [],
-        roles: [:gruppis, :chef],
-        role: :gruppis
-      )
-      |> assign(:active_tab, nil)
-
-    {:ok, socket}
+    user = socket.assigns[:current_user]
+
+    case is_admin?(socket, show_group) do
+      true ->
+        socket =
+          socket
+          |> stream(
+            :members,
+            show_group.group_memberships
+            |> Enum.sort_by(fn %{role: role} -> role end)
+          )
+          |> assign(
+            current_user: user,
+            changeset: changeset,
+            show_group: show_group,
+            page_title: show_group.group.name,
+            query: nil,
+            loading: false,
+            matches: [],
+            roles: [:gruppis, :chef],
+            role: :gruppis
+          )
+
+        {:ok, socket}
+
+      false ->
+        # Redirect to normal group page
+        {:ok, redirect(socket, to: Routes.group_path(socket, :group, show_group.id))}
```

Comment:
This redirect should go to the LiveView group page. Also, it might be nice to display a flash message saying that the user does not have access.